app/models/mixins/purging_mixin.rb

```diff
@@ -46,6 +46,26 @@ def purge_by_date(older_than, window = nil, &block)
       total
     end
 
+    def purge_count_by_remaining(remaining)
+      purge_ids_for_remaining(remaining).size
+    end
+
+    def purge_by_remaining(remaining, window = nil, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...")
+      total = purge_in_batches(purge_ids_for_remaining(remaining), window || purge_window_size, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...Complete - Deleted #{total} records")
+      total
+    end
+
+    def purge_ids_for_remaining(remaining)
```

Comment:
thanks - I changed `id 1 = rank 2 and id 2 = rank 1` (and the other accordingly)
since it is `order id desc`